crates/holochain_types/src/app.rs

```diff
@@ -156,6 +156,33 @@ pub struct InstallAppPayload {
     pub ignore_genesis_failure: bool,
 }
 
+/// Provides default values for [InstallAppPayload] that are not intended to be usable but allow
+/// clients to avoid specifying values for new fields.
+///
+/// ```rust
+/// use std::path::PathBuf;
+/// use holochain_types::app::AppBundleSource;
+/// use holochain_types::prelude::InstallAppPayload;
+/// let payload = InstallAppPayload {
+///     // Provide the fields that are important to your happ install
+///     source: AppBundleSource::Path(PathBuf::from("build/myapp.happ")),
+///     /// Ignore other fields and new fields
+///     ..Default::default()
+/// };
+/// ```
+impl Default for InstallAppPayload {
+    fn default() -> Self {
+        InstallAppPayload {
+            source: AppBundleSource::Path(Default::default()),
+            agent_key: AgentPubKey::from_raw_36(vec![0; 36]),
+            installed_app_id: None,
+            membrane_proofs: Default::default(),
+            network_seed: None,
+            ignore_genesis_failure: false,
+        }
+    }
+}
```

Comment:
Hmm I don't agree with this approach, let's talk about it.

Some of the fields made Default here are not default at all, and it would be quite undesirable if the "default" values were used accidentally. An all-zero agent key, or a default path (what even is that?).

That is questionable enough as it is, but then I'm thinking about future changes. If we have to add yet another field that doesn't have a proper default but must be provided a default anyway, that would be a breaking change in itself: all code that uses default values would not be updated to use a "real" value for that field, the problematic default would be used, and things would probably break as a result.

So to use this approach, we're betting that people are not going to forget to supply the values that are really needed, and that we will only ever add proper "defaulty" fields to this struct in the future. It doesn't seem worth it to me to make that bet.